src/scripts/main.js

```diff
@@ -1,3 +1,324 @@
+  age: parseInt,
+  salary: window.convertSalartyToNumber,
+};
+
+const findSuitableSortingFunction = (
+  index,
+  order,
+  transformFunc = (x) => x,
+) => {
+  return (a, b) => {
+    const valueA = transformFunc(window.getValueByCellIndex(a, index));
+    const valueB = transformFunc(window.getValueByCellIndex(b, index));
+
+    if (typeof valueA === 'string' && typeof valueB === 'string') {
+      return order * valueA.localeCompare(valueB);
+    }
+
+    return order * (valueA - valueB);
+  };
+};
+
+const getSortingFunction = (index, colHeader, ascOrder) => {
+  const transformFunc = valueConversionFunctions[colHeader];
+
+  return ascOrder
+    ? findSuitableSortingFunction(index, 1, transformFunc)
+    : findSuitableSortingFunction(index, -1, transformFunc);
+};
+
+// TOGGLE ASC/DESC ORDER LOGIC
+const ascDescOrderUtil = {
+  element: undefined,
+  ascOrder: undefined, // true -> asc, false -> desc
+
+  clickIsMade: (target) => {
+    if (target === ascDescOrderUtil.element) {
+      ascDescOrderUtil.ascOrder = !ascDescOrderUtil.ascOrder;
+    } else {
+      ascDescOrderUtil.element = target;
+      ascDescOrderUtil.ascOrder = true;
+    }
+  },
+  getValue: () => ascDescOrderUtil.ascOrder,
+};
+
+// HANDLING ASC/DESC CLICK EVENT
+const handleSortingEvent = (e) => {
+  const target = e.target.closest('th');
+
+  if (target) {
+    ascDescOrderUtil.clickIsMade(target);
+
+    const index = target.cellIndex;
+    const colHeader = target.textContent.toLowerCase();
+    const orderDirection = ascDescOrderUtil.getValue();
+
+    Array.from(tBody.rows)
+      .sort(getSortingFunction(index, colHeader, orderDirection))
+      .forEach((row) => tBody.append(row));
+  }
+};
+
+tHead.addEventListener('click', handleSortingEvent);
+// #endregion
+
+// #region SELECT_ROW
+// HANDLING ROW SELECT EVENT
+const deselectRows = () => {
+  for (const row of table.rows) {
+    if (row.classList.contains('active')) {
+      row.classList.remove('active');
+    }
+  }
+};
+
+tBody.addEventListener('click', (e) => {
+  const target = e.target.closest('tr');
+
+  if (target) {
+    deselectRows();
+    target.classList.add('active');
+  }
+});
+// #endregion
+
+// #region FORM_NEW_EMPLOYEE
+// DATA FOR TJE FORM
+const offices = [
+  'Tokyo',
+  'Singapore',
+  'London',
+  'New York',
+  'Edinburgh',
+  'San Francisco',
+];
+
+// CREATING FORM ELEMENTS
+const formatLabelText = (inputName) => {
+  return `${inputName.charAt(0).toUpperCase()}${inputName.substring(1)}: `;
+};
+
+const createInput = (inputName, inputType) => {
+  const input = Object.assign(document.createElement('input'), {
+    name: inputName,
+    type: inputType,
+  });
+  const label = document.createElement('label');
+
+  input.setAttribute('data-qa', inputName);
+
+  label.textContent = formatLabelText(inputName);
+  label.append(input);
+
+  return label;
+};
+
+const createSelect = (inputName, options) => {
+  const select = Object.assign(document.createElement('select'), {
```

Comment:
The `createSelect` function uses `Object.assign` to set properties on a new select element. This is a bit unusual and may be confusing to other developers. Consider using the regular property assignment syntax instead.